src/pipx/package_specifier.py

```diff
@@ -20,6 +20,14 @@
 
 logger = logging.getLogger(__name__)
 
+# Copied from https://github.com/pypa/pip/blob/main/src/pip/_internal/utils/filetypes.py
+WHEEL_EXTENSION = ".whl"
+BZ2_EXTENSIONS: Tuple[str, ...] = (".tar.bz2", ".tbz")
+XZ_EXTENSIONS: Tuple[str, ...] = (".tar.xz", ".txz", ".tlz", ".tar.lz", ".tar.lzma")
+ZIP_EXTENSIONS: Tuple[str, ...] = (".zip", WHEEL_EXTENSION)
+TAR_EXTENSIONS: Tuple[str, ...] = (".tar.gz", ".tgz", ".tar")
+ARCHIVE_EXTENSIONS = ZIP_EXTENSIONS + BZ2_EXTENSIONS + TAR_EXTENSIONS + XZ_EXTENSIONS
```

Comment:
pip supports a ton of files for backward compatibility, but I think we should only support the standard ones: `.whl`, `.tar.gz`, and `.zip`. The wheel extension is obvious, and `packaging` also only supports the two sdist extensions:

https://github.com/pypa/packaging/blob/97db7175675a618b78838ccdf4b78108d3eca6f3/src/packaging/utils.py#L128-L131